core-rust/mesh-api-server/src/mesh_api/handlers/construction_payloads.rs

```diff
@@ -68,62 +83,44 @@ pub(crate) async fn handle_construction_payloads(
         }
     }
     let manifest = builder.build();
-
-    let intent = IntentV1 {
-        header: TransactionHeaderV1 {
-            network_id: state.network.id,
-            start_epoch_inclusive: Epoch::of(metadata.start_epoch_inclusive),
-            end_epoch_exclusive: Epoch::of(metadata.end_epoch_exclusive),
-            nonce: metadata.nonce,
-            notary_public_key: public_key,
-            notary_is_signatory: true,
-            tip_percentage: metadata.tip_percentage,
+    let signed_intent = SignedIntentV1 {
```

Comment:
Out of interest, why didn't we use the TransactionBuilder?

Reply:
`TransactionBuilder` does not have public `signed_transaction_intent_hash()` method.
So we would have to `build()` transaction `NotarizedTransactionV1` to get the required data.
But we cannot notarize it here, because we have no access to private key.